docs/docs/guides/05-type-system/02-concepts/03-public-registry.mdx

```diff
@@ -0,0 +1,316 @@
+import type { Internal, Public } from '@strapi/types'
+
+export type ContentType = Internal.Registry.Keys<
+  Public.ContentTypeSchemas,
+  Internal.UID.ContentType
+>;
+```
+
+:::note
+Only picking-up keys that extend `Internal.UID.ContentType` ensure we don't end up with manually added malformed keys, and tells the type-checker we're confident about what's in our union type.
+:::
+
+###### Schema.ContentType
+
+To declare `Schema.ContentType`, we simply query the content-type schema registry (`Public.ContentTypeSchemas`) with the provided content-type UID (`TUID`).
+
+:::note
+Since `UID.ContentType` (`TUID`) is [dynamically built based on actual keys](#uidcontenttype), we know for sure that there will be a valid corresponding schema in the registry.
+:::
+
+```ts title="@strapi/types/schema.ts"
+import type { UID, Public } from '@strapi/types'
+
+export type ContentType<TUID extends UID.ContentType> = Public.ContentTypeSchemas[TUID];
+```
+
+##### 3. APIs Design
+
+To create types for a dynamic API (_aka one that react to its context_), simply use dynamic type definitions such as `UID`, `Data` or `Schema`.
+
+```ts title="@strapi/core/document-service.ts"
+import type { Data, UID } from '@strapi/types';
+
+export type findOne<TUID extends UID.ContentType>(uid: TUID): Data.ContentType<TUID>;
+```
+
+:::caution
+Remember to use dynamic type definitions (`UID`, `Data`, `Schema`) and not static ones (`Internal`, `Struct`).
+:::
+
+:::info[Reminder]
+Registries are **indexed**, which means that:
+- **When augmented** (_e.g. in users' applications_), they'll return **strongly typed values** that correspond to the defined types.
+- **When empty** (_e.g. in Strapi codebase_), they'll return **generic low level types** based on their index definition.
+
+<Tabs>
+  <TabItem value="augmented" label="Augmented">
+    ```ts
+    import type { UID } from '@strapi/types';
+
+    const uid: UID.ContentType;
+    // ^ 'api::article.article' | 'admin::user'
+    ```
+  </TabItem>
+  <TabItem value="empty" label="Empty">
+    ```ts
+    import type { UID } from '@strapi/types';
+
+    const uid: UID.ContentType;
+    // ^ `admin::${string}` | `api::${string}.${string}` | `plugin::${string}.${string}` | `strapi::${string}`
+    ```
+  </TabItem>
+</Tabs>
+:::
+
+##### 4. Type Augmentation
+
+###### Manual Augmentation
+
+It's possible to manually augment the public registries to create tailor-made experiences.
+
+```ts title="my-app/index.d.ts"
+import type { Struct } from '@strapi/strapi';
+
+interface ApiArticleArticle extends Struct.ContentTypeSchema {
+  // ...
+}
+
+declare module '@strapi/strapi' {
+  export module Public {
+    export interface ContentTypeSchemas {
+      'api::article.article': ApiArticleArticle;
+    }
+  }
+}
+````
+
+This will force every type that depend on the `Public.ContentTypeSchemas` registry to recognize `'api::article.article'` as the only valid UID and `ApiArticleArticle` the only valid schema.
```

Comment:
```suggestion
This will force every type that depends on the `Public.ContentTypeSchemas` registry to recognize `'api::article.article'` as the only valid UID and `ApiArticleArticle` the only valid schema.
```